ddtrace/configuration.py

```diff
@@ -0,0 +1,39 @@
+from collections import defaultdict
+
+
+class ConfigException(Exception):
+    """Configuration exception when an integration that is not available
+    is called in the `Config` object.
+    """
+    pass
+
+
+class Config(object):
+    """Configuration object that exposes an API to set and retrieve
+    global settings for each integration. All integrations must use
+    this instance to register their defaults, so that they're public
+    available and can be updated by users.
+    """
+    def __init__(self):
+        # use a dict as underlying storing mechanism
+        self._config = {}
```

Comment:
This data structure is not "safe" and raises an exception if an integration is not registered. An alternative is to use something like:
```python
self._config = defaultdict(lambda: defaultdict(lambda: None))
self._config.missing_integration['key_that_does_not_exist']  # -> returns None without failing
```

That is a bit "ugly" but does the job. In my opinion, we should fail since global settings must not changed at runtime, though it's something it could happen. This definitely doesn't happen for per instance configs, because they're always available in the `Pin` object.

Reply:
Another option is to return `None` and log a warning in `__getattr__` when you catch a `KeyError` but failing at init time if a tracer was not registered seems the right thing to do here.

Reply:
Yes I think it should work since we expect to fail at init time.